File: src/main/java/org/motivepick/repository/LoginStateRepository.java
package org.motivepick.repository;

import org.motivepick.domain.entity.LoginStateEntity;
import org.springframework.data.repository.PagingAndSortingRepository;

import java.util.Optional;

interface LoginStateRepository extends PagingAndSortingRepository<LoginStateEntity, Long> {

    Optional<LoginStateEntity> findByStateUuid(String uuid);

    Optional<LoginStateEntity> deleteByStateUuid(String uuid);
}

File: src/main/kotlin/org/motivepick/repository/LoginStateRepository.kt
package org.motivepick.repository

import org.motivepick.domain.entity.LoginStateEntity
import org.springframework.data.repository.CrudRepository
import java.util.*

interface LoginStateRepository : CrudRepository<LoginStateEntity, Long> {

    fun findByStateUuid(uuid: String): Optional<LoginStateEntity>

    fun deleteByStateUuid(uuid: String): Optional<LoginStateEntity>
}
